kfs-core/src/main/java/edu/arizona/kfs/gl/batch/BudgetAdjustmentMergeSortStep.java

```diff
@@ -0,0 +1,140 @@
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.SuffixFileFilter;
+import org.kuali.kfs.gl.batch.BatchSortUtil;
+import org.kuali.kfs.gl.batch.ScrubberSortComparator;
+import org.kuali.kfs.sys.batch.AbstractStep;
+
+import edu.arizona.kfs.gl.GeneralLedgerConstants;
+
+public class BudgetAdjustmentMergeSortStep extends AbstractStep {
+	private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(BudgetAdjustmentMergeSortStep.class);
+	
+	protected String batchFileDirectoryName;
+	
+	@Override
+	public boolean execute(String jobName, Date jobRunDate) {
+		String budgetOfficeFileName = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.ENTERPRISE_FEED + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+		String laborLedgerFileName = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.LABOR_GL_ENTRY_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+		String outputFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.BUDGET_ADJUSTMENT_OUTPUT_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
+		
+		// names of the two .done files that we are checking to see if they exist
+		String budgetOfficeDoneFileName = GeneralLedgerConstants.BatchFileSystem.ENTERPRISE_FEED + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+		String laborLedgerDoneFileName = GeneralLedgerConstants.BatchFileSystem.LABOR_GL_ENTRY_FILE + GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION;
+		
+		// pull out a list of the .done files in the batch file directory
+		File directory = new File(batchFileDirectoryName);
+		FileFilter doneFileFilter = new SuffixFileFilter(GeneralLedgerConstants.BatchFileSystem.DONE_FILE_EXTENSION);
+		File[] doneFiles = directory.listFiles(doneFileFilter);
+		
+		boolean laborLedgerDoneFileExists = false;
+		boolean budgetOfficeDoneFileExists = false;
+		
+		// check each file in the list to see if 
+		// the files we need to merge exist in the directory
+		for (File doneFile : doneFiles) {
+			String fileName = doneFile.getName();
+			
+			if (fileName.equals(laborLedgerDoneFileName)) {
+				laborLedgerDoneFileExists = true;
+			} 
+			
+			if (fileName.equals(budgetOfficeDoneFileName)){
+				budgetOfficeDoneFileExists = true;
+			}
+		}
+		
+		// checks to see which files actually exists in the directory
+		// then pulls the necessary .data files to be processed and put into the output file.
+		if (laborLedgerDoneFileExists && budgetOfficeDoneFileExists) {	
+			String mergedFile = batchFileDirectoryName + File.separator + GeneralLedgerConstants.BatchFileSystem.BUDGET_ADJUSTMENT_MERGE_FILE + GeneralLedgerConstants.BatchFileSystem.EXTENSION;
```

Comment:
Add more logging statements to each branch of the else if statement for debugging purposes.